kymatio/scattering1d/core/timefrequency_scattering1d.py

```diff
@@ -0,0 +1,1314 @@
+import math
+from ..backend.agnostic_backend import unpad_dyadic
+
+
+def timefrequency_scattering1d(
+        x, pad, unpad, backend, J, log2_T, psi1, psi2, phi, scf,
+        pad_fn, pad_left=0, pad_right=0, ind_start=None, ind_end=None,
+        oversampling=0, oversampling_fr=0, aligned=True, average=True,
+        average_global=None, average_global_phi=None, out_type='array',
+        out_3D=False, out_exclude=None, pad_mode='zero'):
+    """
+    Main function implementing the Joint Time-Frequency Scattering transform.
+
+    Below is implementation documentation for developers.
+
+    Frequential scattering
+    ======================
+
+    Variables
+    ---------
+    Explanation of variable naming and roles. For `scf`'s attributes see
+    full Attributes docs in `TimeFrequencyScattering1D`.
+
+    subsample_equiv_due_to_pad
+        Equivalent amount of subsampling due to padding, relative to
```

Comment:
Eh, I always found this naming clumsy to first impressions. The docs read:

> Distinguishes differences in input lengths (to `_frequency_scattering()` and `_joint_lowpass()`) due to padding and conv subsampling. This is needed to tell whether input is *trimmed* (pad difference) or *subsampled*.

The idea is, input lengths may differ for two reasons, and for some operations, we need to know exactly for which one. This should be clearer after familiarizing with code. 

For example, `lowpass_subsample_fr` aims for _total_ subsampling to be a certain amount (e.g. `log2_F`). Suppose: input length `32`, target total subsampling `4`. Now, the input could come from: 1) pad to `64`, subsample by `2`; 2) pad to `128`, subsample by `4`. In case 1, we want to subsample further by `2`; in case 2, we're done.

Reply:
I'm open to renaming

Reply:
> For example, lowpass_subsample_fr aims for total subsampling to be a certain amount (e.g. log2_F). Suppose: input length 32, target total subsampling 4. Now, the input could come from: 1) pad to 64, subsample by 2; 2) pad to 128, subsample by 4. In case 1, we want to subsample further by 2; in case 2, we're done.

What you describe sounds like `k1_J` in the 1D code. This logic is not affected by padding/trimming.

I don't see how padding interacts with subsampling (unless you're using array shapes to guess these numbers?)